hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/TestS3AUnbuffer.java

```diff
@@ -67,10 +69,10 @@ public void testUnbuffer() throws IOException {
 
     // Call read and then unbuffer
     FSDataInputStream stream = fs.open(path);
-    assertEquals(0, stream.read(new byte[8])); // mocks read 0 bytes
+    assertEquals(-1, stream.read(new byte[8])); // mocks read 0 bytes
```

Comment:
Please add comment to explain why return value is -1 instead of 0 here

Reply:
that is the correct behavior. see line 62 & 63. read of underlying objectstream returns -1 which should be reflected in input stream read.

Reply:
done